src/main/java/com/google/sps/data/Weather.java

```diff
@@ -0,0 +1,127 @@
+package com.google.sps.data;
+
+import com.google.maps.errors.ApiException;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.lang.InterruptedException;
+import java.io.IOException;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.HttpURLConnection;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.maps.model.LatLng;
+import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonElement;
+
+public class Weather {
+  private Double temperature;
+  private Double temperatureFeelsLike;
+  private Integer pressure;
+  private Integer humidity;
+  private Integer clouds;
+  private String type;
+  private String iconId;
+
+  public Weather() {}
+
+  public Weather(Double temperature, Double temperatureFeelsLike, 
+          Integer pressure, Integer humidity, Integer clouds, 
+          String type, String iconId) {
+    this.temperature = temperature;
+    this.temperatureFeelsLike = temperatureFeelsLike;
+    this.pressure = pressure;
+    this.humidity = humidity;
+    this.clouds = clouds;
+    this.type = type;
+    this.iconId = iconId;
+  }
+
+  private Weather fromOpenWeatherApiJsonObject(JsonObject openWeatherApiJsonObject) {
+    if (openWeatherApiJsonObject == null)
+      return null;
+    JsonElement temperatureElement = openWeatherApiJsonObject.get("temp");
+    Double temperature = null;
+    Double feelsLike = null;
+    if (temperatureElement.isJsonPrimitive()) {
+      temperature = temperatureElement.getAsDouble();
+      feelsLike = openWeatherApiJsonObject.get("feels_like").getAsDouble();
+    } else {
+      temperature = temperatureElement.getAsJsonObject().get("day").getAsDouble();
```

Comment:
What happens if the json object is somehow malformed and doesn't contain some fields like "day" or "feels_like"